pkg/extension/opampextension/opamp_agent.go

```diff
@@ -454,8 +454,8 @@ func (o *opampAgent) saveEffectiveConfig(dir string) error {
 				ProviderFactories: []confmap.ProviderFactory{
 					fileprovider.NewFactory(),
 					yamlprovider.NewFactory(),
+					envprovider.NewFactory(),
 				},
-				ConverterFactories: []confmap.ConverterFactory{expandconverter.NewFactory()},
```

Comment:
I think we should run a test to ensure this doesn't break the opamp agent

Reply:
this converter (expandconverter) was deprecated in v107 (at the bottom of breaking changes): https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.107.0 😆

Reply:
This should be tested in the RC